Classes/Domain/Service/NodePropertyConverterService.php

```diff
@@ -209,15 +214,34 @@ public function getPropertiesArray(Node $node)
     }
 
     /**
-     * Convert the given value to a simple type or an array of simple types.
+     * Convert the given value to a simple type or an array of simple types or to a \JsonSerializable.
      *
-     * @param mixed $propertyValue
-     * @param string $dataType
-     * @return mixed
+     * @param mixed $propertyValue the deserialized node property value
+     * @param string $dataType the property type from the node type schema
+     * @return \JsonSerializable|int|float|string|bool|null|array<mixed>
      * @throws PropertyException
      */
     protected function convertValue($propertyValue, $dataType)
     {
+        if (
+            $propertyValue instanceof \JsonSerializable
+            // todo maybe combine them for performance with a static `isDenormalizable` utility.
+            && (new ValueObjectArrayDenormalizer())->supportsDenormalization([], $propertyValue::class)
```

Comment:
and https://github.com/neos/flow-development-collection/pull/2766 seems to be missing as a feature for the new cr...